src/Microsoft.DotNet.Darc/src/Darc/Models/PopUps/UpdateSubscriptionPopUp.cs

```diff
@@ -0,0 +1,225 @@
+            Contents = new Collection<Line>(new List<Line>
+            {
+                new Line($"Use this form to update the values of subscription '{subscription.Id}'.", true),
+                new Line($"Note that if you are setting 'Is batchable' to true you need to remove all Merge Policies.", true),
+                new Line()
+            });
+
+            foreach (string line in lines)
+            {
+                Contents.Add(new Line(line));
+            }
+
+            // Add helper comments
+            Contents.Add(new Line($"Suggested repository URLs for '{SubscriptionData.sourceRepoElement}':", true));
+
+            foreach (string suggestedRepo in suggestedRepositories)
+            {
+                Contents.Add(new Line($"  {suggestedRepo}", true));
+            }
+
+            Contents.Add(new Line("", true));
+            Contents.Add(new Line("Suggested Channels", true));
+
+            foreach (string suggestedChannel in suggestedChannels)
+            {
+                Contents.Add(new Line($"  {suggestedChannel}", true));
+            }
+
+            Contents.Add(new Line("", true));
+            Contents.Add(new Line("Available Merge Policies", true));
+
+            foreach (string mergeHelp in availableMergePolicyHelp)
+            {
+                Contents.Add(new Line($"  {mergeHelp}", true));
+            }
+        }
+
+        public override int ProcessContents(IList<Line> contents)
+        {
+            SubscriptionData outputYamlData;
+
+            try
+            {
+                string yamlString = contents.Aggregate<Line, string>("", (current, line) => $"{current}{System.Environment.NewLine}{line.Text}");
+                IDeserializer serializer = new DeserializerBuilder().Build();
+                outputYamlData = serializer.Deserialize<SubscriptionData>(yamlString);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Failed to parse input yaml.  Please see help for correct format.");
+                return Constants.ErrorCode;
+            }
+
+            _yamlData.Batchable = ParseSetting(outputYamlData.Batchable, _yamlData.Batchable, false);
+            _yamlData.Enabled = ParseSetting(outputYamlData.Enabled, _yamlData.Enabled, false);
+            // Make sure Batchable and Enabled are valid bools
+            if (!bool.TryParse(outputYamlData.Batchable, out bool batchable) || !bool.TryParse(outputYamlData.Enabled, out bool enabled))
+            {
+                _logger.LogError("Either Batchable or Enabled is not a valid boolean values.");
+                return Constants.ErrorCode;
+            }
+
+            if (!bool.Parse(outputYamlData.Batchable) && outputYamlData.MergePolicies == null)
+            {
+                return Constants.ErrorCode;
+            }
+
+            // Validate the merge policies
+            if (outputYamlData.MergePolicies != null &&!ValidateMergePolicies(outputYamlData.MergePolicies))
```

Comment:
If I remove the merge policies element or just set it to nothing (not empty array):
```
MergePolicies: 
```
This fails to validate and you end up having to use an array `[]` identifier to get it to pass.  We should handle that case because the array syntax is a little non-obvious.

Reply:
Not completely sure I'm following. I tested removing the whole `Merge Policies:` item, removing the contents of `Merge Policies:` and using an empty array `Merge Policies: []` and all three worked fine.

The `!= null` check is to avoid going into `ValidateMergePolicies` since, if it is null, it will NullRef in the foreach loop.